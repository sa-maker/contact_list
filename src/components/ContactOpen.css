.openContactContainer {
  background-color: #ccc;
  padding: 5px 0px 5px 10px;
}
.openContactName {
  font-size: 20px;
  color: #ddd;
}
.openContactDetails {
  text-align: left;
  font-size: 12px;
}
.leftContainer {
  vertical-align: top;
  display: inline-block;
  width: 100px;
  margin-right: 15px;
  margin-top: 8px;
}
.rightContainer {
  display: inline-block;
}
.avatar {
  width: 100px;
  height: 100px;
  object-fit: cover;
  border-radius: 50%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}
.openContactClose {
  float: left;
  font-size: 12px;
  cursor: pointer;
}
.contactUsername {
  transform: rotate(90deg);
  position: relative;
  top: -80px;
  left: 300px;
  font-size: 11px;
  color: #fff;
  padding: 2px 5px;
  width: 160px;
  background-color: red;
}
