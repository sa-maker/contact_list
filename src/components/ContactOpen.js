import React from 'react'
import './ContactOpen.css'

class ContactOpen extends React.Component{
  render() {
    const {contact, setOpenCard} = this.props
    return (
      <div className="openContactContainer">
        <div onClick={()=>setOpenCard('')} className="openContactClose" >X</div>
        <div className="leftContainer">
          <img src={contact.picture.large} alt="Contact profile" className="avatar" />
        </div>
        <div className="rightContainer">  
          <span className="openContactName">  
            {contact.name.last.toUpperCase()}, {contact.name.first}  
          </span>
          <table className="openContactDetails" cellPadding='0px' cellmargin='0px'>
            <tbody>
            <tr>
              <th>e-mail</th>
              <td>{contact.email}</td>
            </tr>
            <tr>
              <th>phone</th>
              <td>{contact.cell}</td>
            </tr>
            <tr>
              <th>street</th>
              <td>{contact.location.street.number} {contact.location.street.name}</td>
            </tr>
            <tr>
              <th>city</th>
              <td>{contact.location.city}</td>
            </tr>
            <tr>
              <th>state</th>
              <td>{contact.location.state}</td>
            </tr>
            <tr>
              <th>postcode</th>
              <td>{contact.location.postcode}</td>
            </tr>
            </tbody>
          </table>
        </div>
        <div className="contactUsername">
            USERNAME {contact.login.username}
        </div>
      </div>
    )      
  }
}

export default ContactOpen