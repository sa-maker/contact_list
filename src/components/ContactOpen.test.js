import React from 'react'
import renderer from 'react-test-renderer'
import ContactOpen from './ContactOpen'

test('Open Card displays props', () => {
    const contact = {
        cell: "06-71-59-27-17",
        dob:
            {
            age: 47,
            date: "1974-02-20T11:49:23.706Z"
            },
        email: "mahe.denis@example.com",
        gender: "male",
        id:
            {
            name: "INSEE",
            value: "1NNaN49601880 02",
            },
        location:
        {
        city: "Poitiers",
        coordinates:
            {
            latitude: "38.6251",
            longitude: "130.8095"
            },
        country: "France",
        postcode: 86468,
        state: "Haute-Corse",
        street:
            {
            name: "Rue de L'Abbé-Groult",
            number: 5576
            },
        timezone:
            {
            description: "Newfoundland",
            offset: "-3:30"
            },
        },
        login:
            {
            md5: "8f1c76c5276a83396cf84c3f2a9ef116",
            password: "1124",
            salt: "8EtZcWrv",
            sha1: "9eee9c7a0fe78b6deaabb91c124b09d5fa6037e1",
            sha256: "02daba5c550410f9752d181bab4c4448d3ce4a17e86f498ca05662a386c956b2",
            username: "lazykoala877",
            uuid: "8cf9b01b-2c35-4350-a118-596393206060"
            },
        name:
            {
            first: "Mahé",
            last: "Denis",
            title: "Mr"
            },
        nat: "FR",
        phone: "01-50-11-99-83",
        picture:
            {
            large: "https://randomuser.me/api/portraits/men/51.jpg",
            medium: "https://randomuser.me/api/portraits/med/men/51.jpg",
            thumbnail: "https://randomuser.me/api/portraits/thumb/men/51.jpg"
            },
        registered:
            {
            age: 18,
            date: "2003-07-16T22:15:49.503Z"
            }
        }
    const setOpenCard = jest.fn()

    const component = renderer.create(
      <ContactOpen contact={contact} setOpenCard={setOpenCard} />  
    )

    let OpenCard = component.toJSON()
    expect(OpenCard).toMatchSnapshot()
  })

