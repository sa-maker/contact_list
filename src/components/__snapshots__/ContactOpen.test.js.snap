// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open Card displays props 1`] = `
<div
  className="openContactContainer"
>
  <div
    className="openContactClose"
    onClick={[Function]}
  >
    X
  </div>
  <div
    className="leftContainer"
  >
    <img
      alt="Contact profile"
      className="avatar"
      src="https://randomuser.me/api/portraits/men/51.jpg"
    />
  </div>
  <div
    className="rightContainer"
  >
    <span
      className="openContactName"
    >
      DENIS
      , 
      Mahé
    </span>
    <table
      cellPadding="0px"
      cellmargin="0px"
      className="openContactDetails"
    >
      <tbody>
        <tr>
          <th>
            e-mail
          </th>
          <td>
            mahe.denis@example.com
          </td>
        </tr>
        <tr>
          <th>
            phone
          </th>
          <td>
            06-71-59-27-17
          </td>
        </tr>
        <tr>
          <th>
            street
          </th>
          <td>
            5576
             
            Rue de L'Abbé-Groult
          </td>
        </tr>
        <tr>
          <th>
            city
          </th>
          <td>
            Poitiers
          </td>
        </tr>
        <tr>
          <th>
            state
          </th>
          <td>
            Haute-Corse
          </td>
        </tr>
        <tr>
          <th>
            postcode
          </th>
          <td>
            86468
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="contactUsername"
  >
    USERNAME 
    lazykoala877
  </div>
</div>
`;
